### DISCUSSION TITLE
Is arm64 supported for diffusers?

### DISCUSSION THREAD

**Discussion by nwcs-sh** [huggingface/diffusers#7585] [Q&A]
I am _not_ talking about Apple silicon.

I have tried to install diffusers on arm systems (ampere), `diffusers["torch"]` seems to clobber the torch install (if using https://catalog.ngc.nvidia.com/orgs/nvidia/containers/pytorch) and returns back errors about missing cuda libraries. When not installing with `torch`, diffusers doesnt seem to use the local pytorch, it also OOMs because it seems to be running off the CPU.

When trying without nvidia container (and just a plain ubuntu container), it seems like diffusers cannot install properly for an arm64 build.

So, the questions:

- Is arm64 CUDA supported?
- Can I use nvcr pytorch with diffusers and diffusers still use CUDA? if so how?

**Comment by tolgacangoz** [comment]
Can I ask which container version are you using? Which versions missing Cuda libraries are they? Also, which docker command did you run?
When installing on just a plain Ubuntu container, what errors does it show?

**Comment by sayakpaul** [comment]
> Is arm64 CUDA supported?

We haven't tested this yet.

**Comment by nwcs-sh** [comment]
After more testing, `diffusers` seems to work with arm64. The versions I used are below.

```
accelerate==0.28.0
diffusers==0.26.3
transformers==4.38.2
```

I was able to get it working with `nvcr.io/nvidia/pytorch:24.02-py3` which has pytorch arm64 image.

**Reply by sayakpaul** [reply]
Thanks for investigating!
